### PR TITLE
feat: separate server related code

### LINKED ISSUE #579: [REFACTOR]: Separate server related code
### Description

This issue seeks to separate server related code into a `server` directory. Code shared between client and server would also be moved to a `shared` directory. This refactor would help to quickly locate code related to each package

### Screenshots

_No response_

### Additional information

_No response_

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

### PR DESCRIPTION
**Pull request by mathiasayivor**
# Fixes Issue

**My PR closes #579**

# 👨‍💻 Changes proposed(What did you do ?)
* Moved server related code to `src/server` directory
* Moved shared code to `src/shared` directory

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

## Note to reviewers

# 📷 Screenshots

### CODE REVIEW

File: src/shared/lib/.gitkeep
Comment (Dun-sin) [inline src/shared/lib/.gitkeep:1]: was this left by mistake?

**Review by Dun-sin — CHANGES_REQUESTED** [review]

File: src/shared/lib/.gitkeep
Comment (mathiasayivor) [reply src/shared/lib/.gitkeep:1]: Nope. I left it there so git could keep that dir.

Comment (Dun-sin) [reply src/shared/lib/.gitkeep:1]: got ya

**Review by Dun-sin — APPROVED** [review]
looks good

**Pull request merged**